state_transfer/load_balancer_endpoint.go

```diff
@@ -0,0 +1,84 @@
+package state_transfer
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type LoadBalancerEndpoint struct {
+	hostname string
+	port     int32
+}
+
+func (r *LoadBalancerEndpoint) createEndpointResources(c client.Client, t Transfer) error {
+	err := createLoadBalancerService(c, r, t)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createLoadBalancerService(c client.Client, r *LoadBalancerEndpoint, t Transfer) error {
+	serviceSelector := labels
+	serviceSelector["pvc"] = t.PVC().Name
```

Comment:
since the label selector will only work when pods have this label. If the name of PVC is larger than 63 chars, this would be an invalid label. We would have to make sure we get the labels truncated or hashed.